apps/homework/templates/homework/submission_detail.html

```diff
@@ -66,30 +83,63 @@
                                         </div>
                                     </div>
 
-                                    <div class="ui grid">
-                                        <div class="column">
-                                            {% for tracker in submission.tracked_submissions.all %}
-                                                {% if tracker.logs %}
-                                                    <div class="ui header">Submission Logs</div>
-                                                    <div class="ui divider"></div>
-                                                    <div class="ui horizontal inline list">
-                                                        {% for log_name in logs_possible %}
-                                                            {% if tracker.logs|has_item:log_name %}
-                                                                <a class="item"
-                                                                   href="{{ tracker.logs|get_item:log_name }}"
-                                                                   target="_blank">
-                                                                    {{ log_name|format_log_name }}
-                                                                </a>
-                                                            {% endif %}
-                                                        {% endfor %}
+                                    {% for tracker in submission.tracked_submissions.all %}
```

Comment:
Load times on this page are getting pretty bad for me. It's taking like 10 seconds to load. Maybe it's time to profile a bit to see what's taking so long. I have a feeling it might be logs, but could be totally wrong.

Reply:
Fixed this. Calling the logs property in the template was making separate calls to that over and over. Just moved it to context so it only gets evaluated once.

Reply:
Much much better. Thanks dude!